data/mods/Xedra_Evolved/spells/hedge_magic_eocs.json

```diff
@@ -538,6 +545,99 @@
       }
     ]
   },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_HEDGE_ASTRAL_PROJECTION",
+    "effect": [
+      { "u_location_variable": { "u_val": "hedge_astral_projection_original_location" }, "min_radius": 0, "max_radius": 0 },
+      { "u_add_effect": "effect_hedge_astral_projection", "duration": "PERMANENT" },
+      { "u_add_effect": "incorporeal", "duration": "PERMANENT" },
+      { "u_message": "As you meditate, you feel your spirit slipping free from your body.", "type": "good" },
+      { "u_add_trait": "ASTRAL_PROJECTION_APPEARANCE" },
+      { "u_spawn_item": "body_astral_projection", "suppress_message": true },
+      { "u_spawn_item": "item_hedge_astral_projection_cord", "suppress_message": true, "force_equip": true },
+      { "math": [ "u_spell_level('hedge_astral_projection_return') = 0" ] },
+      { "run_eocs": "EOC_HEDGE_ASTRAL_PROJECTION_WEARINESS_RECURRING", "time_in_future": 1 }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_HEDGE_ASTRAL_PROJECTION_WEARINESS_RECURRING",
+    "condition": { "u_has_effect": "effect_hedge_astral_projection" },
+    "effect": [
+      { "run_eocs": "EOC_HEDGE_ASTRAL_PROJECTION_WEARINESS_APPLICATION" },
+      { "run_eocs": "EOC_HEDGE_ASTRAL_PROJECTION_WEARINESS_RECURRING", "time_in_future": 1 }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_HEDGE_ASTRAL_PROJECTION_WEARINESS_ON_MOVE",
+    "eoc_type": "EVENT",
+    "required_event": "avatar_moves",
+    "condition": { "u_has_effect": "effect_hedge_astral_projection" },
+    "effect": [
+      { "math": [ "_calorie_amount = rand(8)" ] },
+      { "math": [ "u_calories() -= _calorie_amount" ] },
```

Comment:
A small comment to indicate this is a trick to increase weariness would be helpful to anyone reading this code not knowing the trick.

Reply:
Nevermind, it's in the id.